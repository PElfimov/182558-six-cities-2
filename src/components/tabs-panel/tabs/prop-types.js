import PropTypes from 'prop-types';

export default {
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChangeCity: PropTypes.func.isRequired,
  activeCity: PropTypes.string.isRequired,
};
