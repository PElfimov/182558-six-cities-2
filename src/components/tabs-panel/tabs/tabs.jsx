import React from "react";
import LocationsCity from "../locations-city/locations-city";
import propTypes from "./prop-types";

export default function Tabs(props) {
  const {cities, activeCity, onChangeCity} = props;
  return (
    <div className="tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {cities.map((it, i) => (
            <LocationsCity
              key={it + i}
              name={it}
              isActive={it === activeCity}
              onClick={onChangeCity}
            />
          ))}
        </ul>
      </section>
    </div>
  );
}

Tabs.propTypes = propTypes;
