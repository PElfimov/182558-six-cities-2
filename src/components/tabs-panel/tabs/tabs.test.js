import React from "react";
import renderer from "react-test-renderer";
import {Tabs} from './tabs';

it(`Tabs correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<Tabs
      cities={[`City1`, `City2`, `City3`]}
      onChangeCity={jest.fn()}
      activeCity={`City1`} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
