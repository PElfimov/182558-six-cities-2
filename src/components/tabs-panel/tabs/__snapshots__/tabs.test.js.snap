// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs correctly renders after relaunch 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item  tabs__item--active"
          href="#"
          onClick={[Function]}
        >
          <span>
            City1
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            City2
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            City3
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
